### PR TITLE
feat: date-resolve skill + coordinator scheduling behavior

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- New `date-resolve` skill: deterministic date verification for agents. Call it before writing any date + day-of-week pair — LLMs can't do this arithmetic reliably (the May 7 incident, and my own error during investigation, both prove this)
- `coordinator.yaml`: added explicit warning to call `date-resolve` for date verification, new `## Scheduling Behavior` section (be specific, offer alternatives, verify dates), pinned the skill

## What this fixes

On May 7, ceo-inbox drafted "Monday May 19" for a scheduling reply — May 19 is a Tuesday. The root cause is LLM day-of-week arithmetic failure, which is universal across all models. The fix is a tool rather than a calendar grid: on-demand, zero context waste, handles any date range.

## Test plan

- [ ] `date-resolve` handler tests: 26 passing (`npm test -- --run skills/date-resolve`)
- [ ] Full test suite: 170 files passing (2 pre-existing DB integration failures unrelated to this change)
- [ ] Manual: trigger a scheduling email in the coordinator and verify it calls `date-resolve` before proposing times

## Release Notes

* **New Features**
  * Enhanced coordinator scheduling with explicit date verification requirements
  * Introduced `date-resolve` skill for parsing dates and validating day-of-week pairings; supports absolute dates (multiple formats) and relative expressions

* **Tests**
  * Added comprehensive test coverage for date resolution, including edge cases for month/year boundaries and timezone handling

[Review Change Stack](https://app.coderabbit.ai/change-stack/josephfung/curia/pull/505)

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai review

File: skills/date-resolve/handler.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+    const dayOfWeek = resolved.toFormat('cccc'); // e.g. "Tuesday"
+    const isoDate = resolved.toFormat('yyyy-MM-dd'); // e.g. "2026-05-19"
+    const formatted = resolved.toFormat('cccc, MMMM d, yyyy'); // e.g. "Tuesday, May 19, 2026"
+
+    const result: Record<string, unknown> = {
+      date: isoDate,
+      day_of_week: dayOfWeek,
+      formatted,
+    };
```
[the remark this answers is not in the document]
Comment (josephfung) [reply skills/date-resolve/handler.ts:158]: Fixed in 0b2fcc1 — added `displayTimezone: ctx.timezone ? formatDisplayTimezone(ctx.timezone, new Date()) : null` to the result payload, following the same pattern as `calendar-list-events`, `calendar-find-free-time`, and `held-messages-list`. Also documented in `skill.json` outputs and covered by two new tests.
